code/_helpers/text.dm

```diff
@@ -57,76 +57,116 @@
 /proc/sanitizeSafe(var/input, var/max_length = MAX_MESSAGE_LEN, var/encode = 1, var/trim = 1, var/extra = 1)
 	return sanitize(replace_characters(input, list(">"=" ","<"=" ", "\""="'")), max_length, encode, trim, extra)
 
-//Filters out undesirable characters from names
-/proc/sanitizeName(var/input, var/max_length = MAX_NAME_LEN, var/allow_numbers = 1)
-	if(!input || length(input) > max_length)
-		return //Rejects the input if it is null or if it is longer then the max length allowed
+#define NO_CHARS_DETECTED 0
+#define SPACES_DETECTED 1
+#define SYMBOLS_DETECTED 2
+#define NUMBERS_DETECTED 3
+#define LETTERS_DETECTED 4
 
-	var/number_of_alphanumeric	= 0
-	var/last_char_group			= 0
+/**
+  * Filters out undesirable characters from names.
+  *
+  * * strict - return null immidiately instead of filtering out  TODO: port from tg
+  * * allow_numbers - allows numbers and common special characters - used for silicon/other weird things names
+  */
+
+/proc/sanitizeName(input, max_length = MAX_NAME_LEN, allow_numbers = 0, force_first_letter_uppercase = TRUE)
```

Comment:
`allow_numbers` default was changed from 1 to 0, new `force_first_letter_uppercase` parameter is not provided anywhere nor used in this proc.

Reply:
Changed default value, removed `force_first_letter_uppercase`